src/video/sync/merge.ts

```diff
@@ -0,0 +1,80 @@
+import { tmpdir } from 'node:os';
+import { spawnSync } from 'node:child_process';
+import { mkdtempSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+
+import { Syncer, VideoFile } from './types';
+import { Test } from '@saucelabs/sauce-json-reporter';
+
+/**
+ * MergeSyncer is used to synchronize the video start time of a test case with
+ * a collection of video files. Videos are aggregated and their cumulative
+ * runtime is used to mark the video start time of the next test case to be
+ * added.
+ */
+export class MergeSyncer implements Syncer {
+  duration: number;
```

Comment:
(optional) Maybe worth exporting `Milliseconds` and use it where appropriate? Kinda nice to read it that way!

```suggestion
  duration: Milliseconds;
```